flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/adaptive/CreatingExecutionGraph.java

```diff
@@ -128,10 +129,10 @@ private void handleExecutionGraphCreation(
                                 () ->
                                         StreamSupport.stream(
                                                         executionGraph
-                                                                .getAllExecutionVertices()
+                                                                .getVerticesTopologically()
                                                                 .spliterator(),
                                                         false)
-                                                .map(v -> v.getJobVertex().getJobVertex())
+                                                .map(ExecutionJobVertex::getJobVertex)
                                                 .iterator(),
```

Comment:
Can we reuse `IterableUtils` here to make this slightly more readable?


```
                                        IterableUtils.toStream(
                                                        executionGraph.getVerticesTopologically())
                                                .map(ExecutionJobVertex::getJobVertex)
                                                .iterator(),
```

Reply:
nit: it would be nice to mark `updatedPlan` as final along-side the above change, to make it consistent with rest of the code